From 6fb7ccaadaa38f4e9557641fb05eebd7ad633534 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Sat, 27 Sep 2008 11:13:43 +0800
Subject: [PATCH 022/280] ENGR00093236 VPU: Set IOClkGateSet to true before
 writing register

Set IOClkGateSet to true before writing register in
vpu_DecBitBufferFlush() function.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 3f0ff74..412fe72 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -2380,6 +2380,7 @@ RetCode vpu_DecBitBufferFlush(DecHandle handle)
 		return RETCODE_WRONG_CALL_SEQUENCE;
 	}
 
+	IOClkGateSet(true);
 	if (cpu_is_mx51()) {
 		if (pDecInfo->openParam.bitstreamFormat == STD_DIV3)
 			VpuWriteReg(BIT_RUN_AUX_STD, 1);
@@ -2387,7 +2388,6 @@ RetCode vpu_DecBitBufferFlush(DecHandle handle)
 			VpuWriteReg(BIT_RUN_AUX_STD, 0);
 	}
 
-	IOClkGateSet(true);
 	BitIssueCommand(pCodecInst->instIndex, pCodecInst->codecMode,
 			DEC_BUF_FLUSH);
 
-- 
1.8.0

